genesyscloud/architect_flow/resource_genesyscloud_architect_flow_proxy.go

```diff
@@ -185,3 +297,124 @@ func getAllArchitectFlowsFn(ctx context.Context, p *architectFlowProxy, name str
 
 	return &totalFlows, nil, nil
 }
+
+func generateDownloadUrlFn(a *architectFlowProxy, flowId string) (string, error) {
+	log.Printf("Creating export job for flow %s", flowId)
+	jobId, resp, err := a.createExportJob(flowId)
+	if err != nil {
+		log.Println(err)
+		if resp != nil {
+			log.Printf("API Response: %s", resp.String())
+		}
+		return "", err
+	}
+	log.Printf("Successfully created export job '%s' for flow '%s'", jobId, flowId)
+
+	log.Printf("Polling job '%s' for download url", jobId)
+	downloadUrl, err := a.pollExportJobForDownloadUrl(jobId)
+	if err != nil {
+		return "", err
+	}
+	log.Printf("Successfully read download URL. Export job: %s", jobId)
+
+	return downloadUrl, nil
+}
+
+func createExportJobFn(a *architectFlowProxy, flowId string) (string, *platformclientv2.APIResponse, error) {
```

Comment:
I would have Q generate documentation and unit tests if you have not already written them else where

Reply:
The main method  has all the docs I added a simple comment here also to say that is the implementation function for the method